lib/versioning/Version.js

```diff
@@ -48,6 +59,39 @@ class Version {
     }
 
     /**
+     * Generate a fixed size serialized place holder for deletion (PHD version).
+     * A PHD version is used internally by metadata (VersioningRequestProcessor)
+     * to indicate that the master version of an object has been deleted and it
+     * needs to be updated by the latest version if any.
+     *
+     * @param {string} versionId - the versionId of the version, this expects
+     *                             a fixed length versionId (without info part)
+     *                             to facilitate easy checking if a version is
+     *                             a PHD version; see PHD_VER_LENGTH above
+     * @return {string} - the serialized version
+     */
+    static generatePHDVersion() {
+        return `{ "isPHD": true, "versionId": "${VID.generateVersionId('')}" }`;
+    }
+
+    /**
+     * Put versionId into an object in the (cheap) way of string manipulation,
+     * instead of the more expensive alternative parsing and stringification.
+     *
+     * @param {string} value - stringification of the object to append versionId
+     * @param {string} versionId - the versionId to append
+     * @return {string} - the object with versionId appended
+     */
+    static appendVersionId(value, versionId) {
+        // assuming value has the format of '{...}'
```

Comment:
I'm not sure on this one -> @ttvi?

I think this code is also in the MetaData code as-is. I think we can keep it for post-merge, if there's a sanity check to add.